### PR TITLE
Set Roles When Inviting Users

### LINKED ISSUE #483: Select Roles when Inviting Users
Currently we allow Users to be invited to a Project, making them members when they accept the invitiation.  It was expected that a ProjectCoordinator would then return to the Settings page and set the Roles for that Member.

However, that requires the coordinator to return to that page for each User they add.  Instead lets let a coordinator set the Roles on an Invitation.

### PR DESCRIPTION
**Pull request by ghickman**
This sets up Project invitations to have Roles set when inviting a User to a Project, copying those Roles to the accepted Membership if the User accepts.

It makes use of the RolesForm from #489 and copies the HTML for the Form into the Project settings template (I'd like to abstract that at some point but this is not that point).

Fixes #483

### CODE REVIEW

File: jobserver/templates/project_settings.html [hunk 1]
```diff
@@ -85,27 +86,64 @@ <h3>Invitations</h3>
...
           {% endfor %}
         </select>
 
+        {% for error in form.users.errors %}
+        <p class="text-danger">{{ error }}</p>
+        {% endfor %}
+      </div>
+
+      <div class="form-group mb-3">
+        <p>Select one or more Roles for the User(s) you are inviting.</p>
```
Comment (inglesp) [inline jobserver/templates/project_settings.html:110]: My personal preference is not to capitalise Role and User here, but I'll shut up about this if you like.

File: tests/jobserver/views/test_projects.py [hunk 2]
```diff
@@ -54,6 +54,7 @@ def test_projectacceptinvite_success(rf):
     invite.refresh_from_db()
     assert invite.membership is not None
     assert invite.membership.project == project
+    assert invite.membership.roles == invite.roles
```
Comment (inglesp) [inline tests/jobserver/views/test_projects.py:57]: We never assert anything about what `invite.roles` is.  Is it definitely set to a non-empty list?

**Review by inglesp — APPROVED** [review]
One of the tests could be tightened up, but apart from that, looks good.

File: jobserver/templates/project_settings.html (hunk 1 above)
Comment (ghickman) [reply jobserver/templates/project_settings.html:110]: You're literally the only person keeping me on the straight and narrow with this, please don't stop! 😂

**Pull request merged**